spec/services/withdraw_participant_spec.rb

```diff
@@ -0,0 +1,176 @@
+    end
+  end
+
+  context "when the course identifier is missing" do
+    let(:course_identifier) {}
+
+    it "is invalid and returns an error message" do
+      is_expected.to be_invalid
+
+      expect(service.errors.messages_for(:course_identifier)).to include("The property '#/course_identifier' must be an available course to '#/participant_id'")
+    end
+  end
+
+  context "when the course identifier is an invalid value" do
+    let(:course_identifier) { "invalid-value" }
+
+    it "is invalid and returns an error message" do
+      is_expected.to be_invalid
+
+      expect(service.errors.messages_for(:course_identifier)).to include("The property '#/course_identifier' must be an available course to '#/participant_id'")
+    end
+  end
+
+  context "when the participant identifier is missing" do
+    let(:participant_id) {}
+
+    it "is invalid and returns an error message" do
+      is_expected.to be_invalid
+
+      expect(service.errors.messages_for(:participant_profile)).to include("The property '#/participant_id' must be a valid Participant ID")
+    end
+  end
+
+  context "when the participant identifier is an invalid value" do
+    let(:participant_id) { "invalid-value" }
+
+    it "is invalid and returns an error message" do
+      is_expected.to be_invalid
+
+      expect(service.errors.messages_for(:participant_profile)).to include("The property '#/participant_id' must be a valid Participant ID")
+    end
+  end
+
+  context "when the participant does not belong to the CPD lead provider" do
+    let(:cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider, :with_npq_lead_provider) }
+
+    it "is invalid and returns an error message" do
+      is_expected.to be_invalid
+
+      expect(service.errors.messages_for(:participant_id)).to include("The property '#/participant_id' must be a valid Participant ID")
+    end
+  end
+end
+
+RSpec.shared_examples "validating a participant is not already withdrawn" do
+  it "is invalid and returns an error message" do
+    is_expected.to be_invalid
+
+    expect(service.errors.messages_for(:participant_profile)).to include("The participant is already withdrawn")
+  end
+end
+
+RSpec.shared_examples "withdrawing a participant" do
+  it "creates a participant profile state" do
+    expect { service.call }.to change { ParticipantProfileState.count }
+  end
+
+  it "sets the correct attributes to the new participant profile state" do
+    service.call
+    latest_participant_profile_state = participant_profile.participant_profile_states.order(created_at: :desc).first
+
+    expect(latest_participant_profile_state).to have_attributes(
+      participant_profile_id: participant_profile.id,
+      state: "withdrawn",
+    )
+  end
+
+  it "marks the participant profile as withdrawn" do
+    expect { service.call }.to change { participant_profile.reload.training_status }.from("active").to("withdrawn")
```

Comment:
could you add an expectation for a new induction record being added, since we will be using `Induction::ChangeInductionRecord`?

Reply:
Hi @cwrw tksss 👍 😄  it's been addressed in [51b8090](https://github.com/DFE-Digital/early-careers-framework/pull/2639/commits/51b8090cf4b44933d43fcf8e0ccc693fc7ae0838)